flamingpy/codes/graphs/stabilizer_graph.py

```diff
@@ -289,20 +297,47 @@ def real_nodes(self):
     def real_edges(self):
         """Returns an iterable of all edges excluding the ones connected to the
         'low' or 'high' points."""
-        is_high_or_low = lambda n: n in ("low", "high")
+
+        def is_high_or_low(n):
+            return n in ("low", "high")
+
         return filter(
             lambda edge: not (is_high_or_low(edge[0]) or is_high_or_low(edge[1])),
             self.edges(),
         )
 
+    def assign_weights(self, code):
+        """Assign the weights to the graph based on the weight
+        of the common vertex of each stabilizer pair of the code."""
+        for edge in self.edges():
+            data = self.edge_data(*edge)
+            if data["common_vertex"] is not None:
+                data["weight"] = code.graph.nodes[data["common_vertex"]]["weight"]
+            else:
+                data["weight"] = 0.0
```

Comment:
I'm worried about this. Every edge is constructed from adjacent stabilizers, which must have a common vertex. If there is no common vertex, shouldn't an exception be raised?

Perhaps this is better:
```suggestion
            weight = code.graph.nodes[data["common_vertex"]].get("weight") 
            if weight is not None:
                data["weight"] = weight
```
The `weight = ` line should technically always work. But now, if the weights have not yet been assigned to the EGraph, this simply creates a weightless edge.